app/mailers/notification_mailers/mentioned.rb

```diff
@@ -4,9 +4,10 @@ class Mentioned < NotificationMailers::Base
     delegate :author_name, to: :post, prefix: true
 
     def set_headers(target_id)
-      @post = Mention.find_by_id(target_id).post
+      @post = Mention.find_by_id(target_id).mentions_container
 
       @headers[:subject] = I18n.t('notifier.mentioned.subject', :name => @sender.name)
+      @headers[:template_name] = "mentioned_in_comment" if post.is_a?(Comment)
```

Comment:
I'm fine with both approaches. If you like "split" approach more, I'm fine with implementing it.

However, speaking about deduplication, I see that the workers for Notifications::*  in `app/workers/mail` are almost identical. So are respective methods in `Notifier`. I can refactor them to use some common code instead of deepen the duplication by introducing `Workers::Mail::MentionedInComment`. What do you think?